examples/hello_compute_c/main.c

```diff
@@ -0,0 +1,135 @@
+#include "./../../ffi/wgpu.h"
+#include <stdio.h>
+#include <stdlib.h>
+
+#define WGPU_TARGET_MACOS 1
+#define WGPU_TARGET_LINUX 2
+#define WGPU_TARGET_WINDOWS 3
+
+#define BLEND_STATES_LENGTH (1)
+#define ATTACHMENTS_LENGTH (1)
+#define RENDER_PASS_ATTACHMENTS_LENGTH (1)
+#define BIND_GROUP_LAYOUTS_LENGTH (1)
+
+WGPUByteArray read_file(const char *name) {
+    FILE *file = fopen(name, "rb");
+    fseek(file, 0, SEEK_END);
+    long length = ftell(file);
+    unsigned char *bytes = malloc(length);
+    fseek(file, 0, SEEK_SET);
+    fread(bytes, 1, length, file);
+    fclose(file);
+    return (WGPUByteArray){
+        .bytes = bytes,
+        .length = length,
+    };
+}
+
+void ReadBufferMap(
+	WGPUBufferMapAsyncStatus status, const uint8_t *data, uint8_t *userdata) {
+    if (status == WGPUBufferMapAsyncStatus_Success) {
+        printf("Times: %s \n", data);
+    }
+}
+
+int main(int argc, char *argv[]) {
+
+    if (argc != 5) {
+        return 0;
+    }
+
+    int numbers[] = {
+        atoi(argv[1]),
+        atoi(argv[2]),
+        atoi(argv[3]),
+        atoi(argv[4]),
+    };
+
+    int size = sizeof(numbers);
+
+	uint32_t numbers_length = sizeof(numbers) / sizeof(int);
+
+    WGPUInstanceId instance = wgpu_create_instance();
+
+    WGPUAdapterId adapter = wgpu_instance_get_adapter(instance,
+        &(WGPUAdapterDescriptor){
+            .power_preference = WGPUPowerPreference_LowPower,
+        });
+
+    WGPUDeviceId device = wgpu_adapter_request_device(adapter,
+        &(WGPUDeviceDescriptor){
+            .extensions =
+                {
+                    .anisotropic_filtering = false,
+                },
+        });
+
+    WGPUBufferId staging_buffer = wgpu_device_create_buffer_mapped(device,
```

Comment:
`wgpu_device_create_buffer_mapped` accepts a `uint8_t ** mapped_ptr_out`. `mapped_ptr_out` is a pointer to some mapped memory for the buffer. After the mapped memory has been updated, you should use `wgpu_buffer_unmap` to finish updating the memory for the buffer.

So, instead of passing `numbers` here directly, we need to use `mapped_ptr_out` and copy the numbers into that memory instead, and use `wgpu_buffer_unmap` afterwards.

Reply:
I am not exactly sure about what you mean with this, maybe it is my lack in C, but do I have to create another variable that hold the buffer, and then pass that buffer into the function?